gh-116738: Make _heapq module thread-safe (#135036)
python/cpython: Misc/NEWS.d/next/Core_and_Builtins/2025-06-02-13-57-40.gh-issue-116738.ycJsL8.rst

> Make methods on :class:`heapq` thread-safe when the GIL is disabled.

Reviewer: ```suggestion
Make methods in :mod:`heapq` thread-safe on the :term:`free threaded <free threading>` build.
```
